mysql-replicator-streams/src/main/java/com/booking/replication/streams/StreamsImplementation.java

```diff
@@ -112,6 +117,11 @@
         this.handler = (exception) -> StreamsImplementation.LOG.error("error inside streams", exception);
     }
 
+    public void registerMetric(MetricRegistry metricRegistry) {
```

Comment:
MetricRegistry is already singleton. So we dont need to initialise it again here. Me can just call Metrics.getInstance(configuration). Please refer to its usage on com.booking.replication.augmenter.AugmenterContext. So that we can completely remove these methods from the base interface. I can help you with changes.